services/common/src/components/projectSummary/AuthorizationsInvolved.tsx

```diff
@@ -37,16 +37,88 @@ import { fetchPermits } from "@mds/common/redux/actionCreators/permitActionCreat
 import { createDropDownList } from "@mds/common/redux/utils/helpers";
 import { FORM } from "@mds/common/constants/forms";
 import RenderHiddenField from "../forms/RenderHiddenField";
+import AuthorizationSupportDocumentUpload from "./AuthorizationSupportDocumentUpload";
+import { IProjectSummaryDocument } from "@mds/common";
+import {
+  renderTextColumn,
+  renderDateColumn,
+} from "@mds/common/components/common/CoreTableCommonColumns";
+import DocumentTable from "@mds/common/components/documents/DocumentTable";
+import {
+  PROJECT_SUMMARY_DOCUMENT_TYPE,
+  PROJECT_SUMMARY_DOCUMENT_TYPE_CODE,
+  PROJECT_SUMMARY_DOCUMENT_TYPE_CODE_LOCATION,
+} from "../..";
+import { renderCategoryColumn } from "@mds/common/components/common/CoreTableCommonColumns";
+import { MineDocument } from "@mds/common/models/documents/document";
+
+export interface ProjectSummary {
+  project_summary_id: number;
+  mine_guid: string;
+  project_summary_guid: string;
+  status_code: string;
+  submission_date: string;
+  project_summary_description: string;
+  project_guid: string;
+  documents: IProjectSummaryDocument[];
+}
 
-const RenderEMAPermitCommonSections = ({ isAmendment }) => {
-  const purposeLabel = isAmendment
+interface DocumentUploadProps {
+  initialValues: ProjectSummary;
+  change: any;
+  documents: IProjectSummaryDocument[];
+  isEditMode: boolean;
+  projectSummaryDocumentTypesHash: any;
+  mineGuid: string;
+}
+
+const RenderEMAPermitCommonSections = ({ props }) => {
+  const dispatch = useDispatch();
+  const purposeLabel = props.isAmendment
     ? "Additional Amendment Request Information"
     : "Purpose of Application";
-  const [showDocSection, setShowDocSection] = useState(false);
+
+  const [showDocSection, setShowDocSection] = useState(props?.exemption_requested || false);
 
   const onChange = (value, _newVal, _prevVal, _fieldName) => {
     setShowDocSection(value);
   };
+
+  const updateAmendmentDocuments = (doc: IProjectSummaryDocument) => {
+    const index = 0;
+    const response = dispatch(
+      arrayPush(
+        FORM.ADD_EDIT_PROJECT_SUMMARY,
+        `authorizations.AIR_EMISSIONS_DISCHARGE_PERMIT.AMENDMENT[${index}].amendment_documents`,
+        doc
+      )
+    );
+    return response;
+  };
+
+  const tableDocuments =
+    props?.amendment_documents?.map(
+      (doc) => new MineDocument({ ...doc, category: doc.project_summary_document_type_code })
+    ) ?? [];
+
+  const documentColumns = [
+    renderTextColumn("document_name", "File Name"),
+    renderCategoryColumn(
+      "project_summary_document_type_code",
```

Comment:
This function doesn't seems rendering the file type as expected. I tried `projectSummaryDocumentTypesHash` from the props initially and then tried passing a map as above but nothing worked. And also noticed that its not working in other [document-upload table in the form](https://github.com/bcgov/mds/blob/206487d16e9d80ac1b816d37240e743ff5137bc8/services/minespace-web/src/components/Forms/projects/projectSummary/DocumentUpload.tsx#L107).

Did some debugging and its complaining about `text is undefined` [in the function here](https://github.com/bcgov/mds/blob/206487d16e9d80ac1b816d37240e743ff5137bc8/services/common/src/components/common/CoreTableCommonColumns.tsx#L62)

Reply:
Took a look.  The document records in the table look like this:

```
{
    "mine_document_guid": "c2b1b949-26e2-45a2-9f20-160851a174fd",
    "mine_guid": "34af5a68-e1f6-47c8-9aa8-ec4b49180e1d",
    "document_manager_guid": "3730a84c-4d76-4238-b067-fd644e543914",
    "document_name": "test1.pdf",
    "create_user": "idir\\matbusby-ey@bceid",
    "upload_date": "2024-05-06 17:37:15.986560+00:00",
    "update_timestamp": null,
    "category": "MAP",
    "is_archived": false,
    "archived_by": null,
    "archived_date": null,
    "is_latest_version": true,
    "entity_title": "",
    "key": "c2b1b949-26e2-45a2-9f20-160851a174fd",
    "file_type": ".pdf",
    "number_prev_versions": 0,
    "versions": [],
    "allowed_actions": [
        "Open in document viewer",
        "Download file",
        "Replace file",
        "Archive file",
        "Delete"
    ]
}
```

The first property you pass to `renderCategoryColumn` will be the data index which has to reference the data within the record object.  So, since the record simply has a `category` field rather than a `project_summary_document_type_code`, changing that first prop to `category` should do the trick.

Reply:
oh, I was playing with the map. Thanks. Updated accordingly and now it works.